packages/neuron-ui/src/widgets/DropdownWithCustomRender/index.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { ReactNode, useEffect, useMemo, useRef, useState } from 'react'
+import { LineDownArrow } from 'widgets/Icons/icon'
+
+import styles from './index.module.scss'
+
+export interface OptionProps {
+  label: string | ReactNode
+  value: string
+  className?: string
+  data?: Record<string, string | number>
+}
+
+interface DropdownWithCustomRenderProps {
+  onChange: (changeValue: OptionProps) => void
+  options: OptionProps[]
+  value?: string
+  placeholder?: string
+  onFocus?: (isOpen: boolean) => void
+  disabled?: boolean
+  className?: string
+}
+
+const DEFAULT_PLACEHOLDER_STRING = 'Please select...'
+
+const DropdownWithCustomRender = ({
+  options: optionsFromProps,
+  onChange,
+  value: valueFromProps = '',
+  placeholder,
+  onFocus,
+  disabled,
+  className,
+}: DropdownWithCustomRenderProps) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [isMounted, setIsMounted] = useState<boolean>(true)
+  const dropdownRef = useRef<HTMLDivElement | null>(null)
+
+  // render selected value in options
+  const parseValue = (value: string, options: OptionProps[]) => {
+    let option
+
+    for (let i = 0, num = options.length; i < num; i++) {
+      if (typeof options[i].value !== 'undefined' && options[i].value === value) {
+        option = options[i]
+      }
+    }
+
+    return (
+      option || {
+        label: typeof placeholder === 'undefined' ? DEFAULT_PLACEHOLDER_STRING : placeholder,
+        value: '',
+      }
+    )
+  }
+
+  const [selectedValue, setSelectedValue] = useState<OptionProps>(
+    parseValue(valueFromProps, optionsFromProps) || {
```

Comment:
`parseValue` will not return undefined, it may be no need to add `||`

Reply:
already remove `| |`